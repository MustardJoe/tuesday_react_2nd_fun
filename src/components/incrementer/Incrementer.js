import React, { Component } from 'react';

class Incrementer extends Component {
  state = {
    count: 0,
    otherThing: 'happyDay',
  };

  incrementCount = () => {
    this.setState(state => {
      return { count: state.count + 1 };
    });
  }

  render() {
    const { count, otherThing } = this.state;

    return (
      <>
        <p>Current count: {count}</p>
        <p>{otherThing}</p>
        <button onClick={this.incrementCount}>Make go up by one!</button>
      </>
    );
  }
}

export default Incrementer;
