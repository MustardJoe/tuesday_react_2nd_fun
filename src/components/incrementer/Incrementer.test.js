import React from 'react';
import { shallow } from 'enzyme';
import Incrementer from './Incrementer';

describe('Incrementer component', () => {
  it('renders Incrementer', () => {
    const wrapper = shallow(<Incrementer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('the button works for to go up by one (increment)', () => {
    const wrapper = shallow(<Incrementer />);

    wrapper.find('button').at(0).simulate('click');
    expect(wrapper.state('count')).toEqual(1);
  });
});
