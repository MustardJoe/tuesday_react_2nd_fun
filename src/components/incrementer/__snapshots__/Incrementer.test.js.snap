// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incrementer component renders Incrementer 1`] = `
<Fragment>
  <p>
    Current count: 
    0
  </p>
  <p>
    happyDay
  </p>
  <button
    onClick={[Function]}
  >
    Make go up by one!
  </button>
</Fragment>
`;
